infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -91,17 +98,81 @@ public LogReplicationConfig(Set<String> streamsToReplicate, int maxNumMsgPerBatc
     /**
      * Constructor
      *
-     * @param streamsToReplicate Unique identifiers for all streams to be replicated across sites.
+     * @param streamsToReplicate Info for all streams to be replicated across sites
      * @param maxNumMsgPerBatch snapshot sync batch size (number of entries per batch)
      */
-    public LogReplicationConfig(Set<String> streamsToReplicate, int maxNumMsgPerBatch, int maxMsgSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, MAX_CACHE_NUM_ENTRIES);
+    public LogReplicationConfig(Set<TableInfo> streamsToReplicate, int maxNumMsgPerBatch, int maxMsgSize) {
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, MAX_CACHE_NUM_ENTRIES, null);
     }
 
-    public LogReplicationConfig(Set<String> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
-                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize, int cacheSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize);
+    public LogReplicationConfig(Set<TableInfo> streamsToReplicate, Map<UUID, List<UUID>> streamingTagsMap,
+                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize,
+                                int cacheSize, LogReplicationStreamNameTableManager streamInfoManager) {
+        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize, streamInfoManager);
         this.dataStreamToTagsMap = streamingTagsMap;
         this.mergeOnlyStreams = mergeOnlyStreams;
     }
-}
\ No newline at end of file
+
+    /**
+     * This class represents Log Replication streams info
+     */
+    public static class StreamInfo {
+        private final Set<UUID> streamIds;
+        private final LogReplicationStreamNameTableManager streamInfoManager;
+
+        public StreamInfo(@NonNull Set<TableInfo> streamsToReplicate,
+                          LogReplicationStreamNameTableManager streamInfoManager) {
+            this.streamInfoManager = streamInfoManager;
+            this.streamIds = new HashSet<>();
+            refreshStreamIds(streamsToReplicate, true);
+        }
+
+        private void refreshStreamIds(Set<TableInfo> streamsToReplicate, boolean shouldClear) {
```

Comment:
I saw all reference of this refreshStreamIds function has a shouldClear set to true, do we really need this flag here?